src/python/pants/engine/internals/build_files_test.py

```diff
@@ -1,57 +1,47 @@
 # Copyright 2015 Pants project contributors (see CONTRIBUTORS.md).
```

Comment:
Summary of the changes:
- Removed the JSON parser examples. Those were testing functionality that we don't use in production, like strong references to other structs. Replaced with some integration-style tests for BUILD file parsing.
- Rewrote tests from unittest to Pytest style.
- Added new tests for finding `BuildFileAddress` and for `AddressesWithOrigins -> Addresses`.